v2/internal/controllers/crd_networking_privateendpoint_test.go

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package controllers_test
+
+import (
+	"testing"
+
+	"github.com/Azure/go-autorest/autorest/to"
+	. "github.com/onsi/gomega"
+
+	"github.com/Azure/azure-service-operator/v2/api/network/v1beta20201101"
+	network "github.com/Azure/azure-service-operator/v2/api/network/v1beta20220701"
+	resources "github.com/Azure/azure-service-operator/v2/api/resources/v1beta20200601"
+	storage "github.com/Azure/azure-service-operator/v2/api/storage/v1beta20210401"
+	"github.com/Azure/azure-service-operator/v2/internal/testcommon"
+)
+
+func Test_Networking_PrivateEndpoint_CRUD(t *testing.T) {
+	t.Parallel()
+
+	tc := globalTestContext.ForTest(t)
+
+	rg := tc.CreateTestResourceGroupAndWait()
+
+	saKind := storage.StorageAccount_Kind_Spec_BlobStorage
+	sa := newStorageAccount(tc, rg)
+	sa.Spec.Kind = &saKind
+
+	vnet := newVMVirtualNetwork(tc, testcommon.AsOwner(rg))
+	subnet := newVMSubnet(tc, testcommon.AsOwner(vnet))
+
+	tc.CreateResourcesAndWait(sa, vnet, subnet)
+
+	tc.Expect(sa.Status.Id).ToNot(BeNil())
+
+	endpoint := newPrivateEndpoint(tc, rg, sa, subnet)
+
+	tc.CreateResourceAndWait(endpoint)
+
+	tc.Expect(endpoint.Status.Id).ToNot(BeNil())
+	armId := *endpoint.Status.Id
+
+	old := endpoint.DeepCopy()
+	key := "foo"
+	endpoint.Spec.Tags = map[string]string{key: "bar"}
+
+	tc.PatchResourceAndWait(old, endpoint)
+	tc.Expect(endpoint.Status.Tags).To(HaveKey(key))
+
+	tc.RunParallelSubtests(
+		testcommon.Subtest{
+			Name: "Test_DNSZoneGroup_CRUD",
+			Test: func(tc *testcommon.KubePerTestContext) {
+				DNSZoneGroup_CRUD(tc, vnet, endpoint, rg)
+			},
+		},
+	)
+
+	tc.DeleteResourceAndWait(endpoint)
+
+	// Ensure delete
+	exists, retryAfter, err := tc.AzureClient.HeadByID(tc.Ctx, armId, string(network.APIVersion_Value))
+	tc.Expect(err).ToNot(HaveOccurred())
+	tc.Expect(retryAfter).To(BeZero())
+	tc.Expect(exists).To(BeFalse())
+
+}
+
+func DNSZoneGroup_CRUD(tc *testcommon.KubePerTestContext, vnet *v1beta20201101.VirtualNetwork, endpoint *network.PrivateEndpoint, rg *resources.ResourceGroup) {
```

Comment:
Ok so this test is basically testing the scenario where the user has a VNET + PrivateDNSZone already configured on that VNET, and they want the PrivateEndpoint to create an entry in that existing PrivateDNSZone?

This seems pretty OK from a user-experience for me.

What about the other scenarios? User has NO private DNS zone and just wants to associate the Private Endpoint with a VNET directly. Should we have another test for that scenario? That scenario will make use of the `customDnsConfigs` property of the `PrivateEndpoint`, right?